src/NServiceBus.AcceptanceTests/Sagas/When_saga_handles_messages_concurrently.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_saga_handles_messages_concurrently : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public Task Should_complete_saga_eventually()
```

Comment:
the naming somehow doesn't really match the specific expectations. How about something like
`When_saga_started_concurrently` - `Should_start_single_saga` ?